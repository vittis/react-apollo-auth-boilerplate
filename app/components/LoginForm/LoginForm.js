import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  Container,
  Col,
  Form,
  FormGroup,
  Label,
  Input,
  Button,
  Alert,
} from 'reactstrap';
import './loginForm.module.scss';

// TODO: Add prop validation
/* eslint-disable react/prefer-stateless-function */
class LoginForm extends Component {
  render() {
    const { email, password } = this.props;
    return (
      <Container styleName="App">
        <h2>Sign In</h2>
        <Form styleName="form" onSubmit={e => this.props.onSubmit(e)}>
          <Col>
            <FormGroup>
              <Label>Username</Label>
              <Input
                required
                type="text"
                name="email"
                id="exampleEmail"
                placeholder="myemail@email.com"
                value={email}
                onChange={e => {
                  this.props.onChangeHandle(e);
                }}
              />
            </FormGroup>
          </Col>
          <Col>
            <FormGroup>
              <Label for="examplePassword">Password</Label>
              <Input
                required
                type="password"
                name="password"
                id="examplePassword"
                placeholder="********"
                value={password}
                onChange={e => this.props.onChangeHandle(e)}
              />
            </FormGroup>
          </Col>
          {this.props.loading && <Alert color="info">Loading...</Alert>}
          {this.props.error && (
            <Alert color="danger">Could not login. Please try again.</Alert>
          )}

          <Button disabled={this.props.loading}>Submit</Button>
        </Form>
      </Container>
    );
  }
}
LoginForm.propTypes = {
  email: PropTypes.string,
  password: PropTypes.string,
  onSubmit: PropTypes.func,
  onChangeHandle: PropTypes.func,
  error: PropTypes.object,
  loading: PropTypes.bool,
};
export default LoginForm;
