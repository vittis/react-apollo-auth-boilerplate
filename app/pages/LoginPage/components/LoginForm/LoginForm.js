import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  Container,
  Col,
  Form,
  FormGroup,
  Label,
  Input,
  Button,
  Alert,
} from 'reactstrap';
import './loginForm.module.scss';

/* eslint-disable react/prefer-stateless-function */
class LoginForm extends Component {
  render() {
    const { email, password, error, loading, called } = this.props;

    const shouldRenderLoadingAlert = loading && called;
    const shouldRenderErrorAlert = error;
    const shouldRenderSucessAlert = called && !loading && !error;

    return (
      <Container styleName="App">
        <h2>Sign In</h2>
        <Form styleName="form" onSubmit={e => this.props.onSubmit(e)}>
          <Col>
            <FormGroup>
              <Label>Username</Label>
              <Input
                required
                type="text"
                name="email"
                id="exampleEmail"
                placeholder="myemail@email.com"
                value={email}
                onChange={e => {
                  this.props.onChangeHandle(e);
                }}
              />
            </FormGroup>
          </Col>
          <Col>
            <FormGroup>
              <Label for="examplePassword">Password</Label>
              <Input
                required
                type="password"
                name="password"
                id="examplePassword"
                placeholder="********"
                value={password}
                onChange={e => this.props.onChangeHandle(e)}
              />
            </FormGroup>
          </Col>

          {shouldRenderLoadingAlert && <Alert color="info">Loading...</Alert>}
          {shouldRenderErrorAlert && (
            <Alert color="danger">Could not login. Please try again.</Alert>
          )}
          {shouldRenderSucessAlert && (
            <Alert color="sucess">Sucess! You should be redirected soon.</Alert>
          )}

          <Button disabled={this.props.loading}>Submit</Button>
        </Form>
      </Container>
    );
  }
}
LoginForm.propTypes = {
  email: PropTypes.string.isRequired,
  password: PropTypes.string.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onChangeHandle: PropTypes.func.isRequired,
  error: PropTypes.bool.isRequired,
  loading: PropTypes.bool.isRequired,
  called: PropTypes.bool.isRequired,
};
export default LoginForm;
