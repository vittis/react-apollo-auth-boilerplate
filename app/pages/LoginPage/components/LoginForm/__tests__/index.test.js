import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import LoginForm from '../index';

const props = {
  email: '',
  password: '',
  onSubmit: jest.fn(),
  onChangeHandle: jest.fn(),
  error: false,
  loading: false,
  called: false,
};

describe('<LoginForm />', () => {
  it('should render and match snapshot', () => {
    const renderedComponent = shallow(
      <LoginForm
        email={props.email}
        password={props.password}
        onSubmit={props.onSubmit}
        onChangeHandle={props.onChangeHandle}
        error={props.error}
        loading={props.loading}
        called={props.called}
      />
    );
    expect(toJSON(renderedComponent)).toMatchSnapshot();
  });
  it('should render root component', () => {
    const renderedComponent = shallow(
      <LoginForm
        email={props.email}
        password={props.password}
        onSubmit={props.onSubmit}
        onChangeHandle={props.onChangeHandle}
        error={props.error}
        loading={props.loading}
        called={props.called}
      />
    );

    expect(renderedComponent.length).toEqual(1);
  });

  it('should render two inputs', () => {
    const renderedComponent = shallow(
      <LoginForm
        email={props.email}
        password={props.password}
        onSubmit={props.onSubmit}
        onChangeHandle={props.onChangeHandle}
        error={props.error}
        loading={props.loading}
        called={props.called}
      />
    );
    expect(renderedComponent.find('Input').length).toEqual(2);
  });

  it('should render email and password in input via props', () => {
    const renderedComponent = shallow(
      <LoginForm
        email={props.email}
        password={props.password}
        onSubmit={props.onSubmit}
        onChangeHandle={props.onChangeHandle}
        error={props.error}
        loading={props.loading}
        called={props.called}
      />
    );
    expect(
      renderedComponent
        .find('Input')
        .first()
        .props().value
    ).toEqual(props.email);
    expect(
      renderedComponent
        .find('Input')
        .last()
        .props().value
    ).toEqual(props.password);
  });
  it('should render error alert if an error occurs', () => {
    props.error = true;
    const renderedComponent = shallow(
      <LoginForm
        email={props.email}
        password={props.password}
        onSubmit={props.onSubmit}
        onChangeHandle={props.onChangeHandle}
        error={props.error}
        loading={props.loading}
        called={props.called}
      />
    );
    expect(renderedComponent.find('Alert').length).toEqual(1);
  });
  it('should render loading alert when loading', () => {
    props.error = false;
    props.loading = true;
    props.called = true;
    const renderedComponent = shallow(
      <LoginForm
        email={props.email}
        password={props.password}
        onSubmit={props.onSubmit}
        onChangeHandle={props.onChangeHandle}
        error={props.error}
        loading={props.loading}
        called={props.called}
      />
    );
    expect(renderedComponent.find('Alert').length).toEqual(1);
  });
  it('should render sucess alert when no errors occurs', () => {
    props.error = false;
    props.loading = false;
    props.called = true;
    const renderedComponent = shallow(
      <LoginForm
        email={props.email}
        password={props.password}
        onSubmit={props.onSubmit}
        onChangeHandle={props.onChangeHandle}
        error={props.error}
        loading={props.loading}
        called={props.called}
      />
    );
    expect(renderedComponent.find('Alert').length).toEqual(1);
  });
});
