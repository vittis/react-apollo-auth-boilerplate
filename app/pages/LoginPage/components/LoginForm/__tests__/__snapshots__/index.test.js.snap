// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> should render and match snapshot 1`] = `
<Container
  className="loginForm-module--App-3ruTd"
  tag="div"
>
  <h2>
    Sign In
  </h2>
  <Form
    className="loginForm-module--form-l2wxs"
    onSubmit={[Function]}
    tag="form"
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <FormGroup
        tag="div"
      >
        <Label
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          Username
        </Label>
        <Input
          id="exampleEmail"
          name="email"
          onChange={[Function]}
          placeholder="myemail@email.com"
          required={true}
          type="text"
          value=""
        />
      </FormGroup>
    </Col>
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <FormGroup
        tag="div"
      >
        <Label
          for="examplePassword"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          Password
        </Label>
        <Input
          id="examplePassword"
          name="password"
          onChange={[Function]}
          placeholder="********"
          required={true}
          type="password"
          value=""
        />
      </FormGroup>
    </Col>
    <Button
      color="secondary"
      disabled={false}
      tag="button"
    >
      Submit
    </Button>
  </Form>
</Container>
`;
